codec-http2/src/main/java/io/netty/handler/codec/http2/StreamBufferingEncoder.java

```diff
@@ -159,9 +168,9 @@ public ChannelFuture writeHeaders(ChannelHandlerContext ctx, int streamId, Http2
             return super.writeHeaders(ctx, streamId, headers, streamDependency, weight,
                     exclusive, padding, endOfStream, promise);
         }
-        if (goAwayLastStreamId != null) {
+        if (goAwayDetail != null) {
             promise.setFailure(new Http2GoAwayException(
-                goAwayLastStreamId, goAwayErrorCode, ByteBufUtil.getBytes(goAwayDebugData)));
+                goAwayDetail.lastStreamId, goAwayDetail.lastStreamId, goAwayDetail.debugData));
```

Comment:
Just pass the `goAwayDetail` as the lone argument? I don't really care much, but it just was a different approach than in cancelGoAwayStreams.